extensions/ql-vscode/src/codeql-task-provider.ts

```diff
@@ -0,0 +1,111 @@
+import * as vscode from 'vscode';
+import { CodeQLCliServer } from './cli';
+
+interface CodeQLTaskDefinition extends vscode.TaskDefinition {
+  /**
+   * The command to run.
+   */
+  command: string;
+
+  /**
+   * Additional command arguments.
+   */
+  commandArgs?: string[];
+}
+
+export class CodeQLTaskProvider implements vscode.TaskProvider {
+  static CodeQLType = 'codeql';
+  private tasks: vscode.Task[] | undefined;
+
+  constructor(private cliServer: CodeQLCliServer) { }
+
+  public provideTasks(): vscode.Task[] {
+    return this.getTasks();
+  }
+
+  public resolveTask(_task: vscode.Task): vscode.Task | undefined {
+    const command: string = _task.definition.command;
+    if (command) {
+      const definition: CodeQLTaskDefinition = <any>_task.definition;
+      return this.getTask(definition.command, definition.commandArgs ?? [], definition);
+    }
+    return undefined;
+  }
+
+  private getTasks(): vscode.Task[] {
+    if (this.tasks !== undefined) {
+      return this.tasks;
+    }
+    // Hard-code examples for now. Not sure if we can do this in a better way.
+    const commands: string[] = [
+      'resolve languages',
+      'version',
+      'pack install',      // TODO: Specify which qlpack to install dependencies for.
+      'resolve queries',   // TODO: Specify which query, directory, or suite to resolve. Also specify search path/additional packs.
```

Comment:
Hmmm...not sure if/how we can specify this. For `pack install`, I think you can look at the active editor (or selected file in the side bar) and assume the qlpack is in that directory, but that is a bit of a pain/

Maybe easier to put up select box where user can choose which pack to run on.

Not sure about `resolve queries`. I think it is less important than the other commands.